src/hooks/useDropzoneArea.js

```diff
@@ -0,0 +1,80 @@
+import {useState, useEffect, useCallback} from 'react';
+import {createFileFromUrl, readFile} from '../helpers';
+
+export const useDropzoneArea = ({
+    onChange,
+    clearOnUnmount,
+    initialFiles,
+    filesLimit,
+    onDelete}) => {
+    const [fileObjects, setFileObjects] = useState([]);
+
+    useEffect(() => {
+        if (onChange) {
+            onChange(fileObjects.map((fileObject) => fileObject.file));
+        }
+    }, [fileObjects, onChange]);
+
+    useEffect(() => {
+        loadInitialFiles();
+
+        return () => {
+            if (clearOnUnmount) {
+                setFileObjects([]);
+            }
+        };
+    }, [loadInitialFiles]);
+
+    const loadInitialFiles = useCallback(async() => {
+        try {
+            const fileObjs = await Promise.all(
+                initialFiles.map(async(url) => {
+                    const file = await createFileFromUrl(url);
+                    const data = await readFile(file);
+                    return {
+                        file,
+                        data,
+                    };
+                })
+            );
+
+            setFileObjects((prev) => ([...prev, ...fileObjs]));
+        } catch (err) {
+            console.log(err);
+        }
+    }, [initialFiles]);
+
+    const addFiles = useCallback(async(newFileObjects) => {
+    // Update component state
+        setFileObjects((prev) => {
+            if (filesLimit <= 1) {
+                return [newFileObjects[0]];
+            }
+
+            return [...prev, ...newFileObjects];
+        });
+    }, [filesLimit]);
+
+    const deleteFile = useCallback((removedFileObj, removedFileObjIdx) => {
```

Comment:
Seems a little weird to me to pass in the object and index. This just makes it possible to mismatch these. It'd be nicer to pass in one or the other (or a unique identifier) and handle the filtering of deleted/not deleted in this method which is already half happening.

Reply:
Perhaps, although this was existing logic that I just extracted

Reply:
I think it might be a good idea to have unit tests for some of these functions, that way we can be confidant that when we refactor them we haven't broken them, @panz3r  what do you think about introducing tests such as jest? Also what about storybook? That way we can have more concise range of examples in the docs and a more verbose set of examples for developing against in the story book. If you think this is a good idea @panz3r , I can create an issue for this, then we can start going through and refactoring bits of code like this

Reply:
@max-carroll ,

> what do you think about introducing tests such as jest?

This is definitely on the roadmap I'd say, I'd really love to have some tests around. 
Using `jest` and `react-testing-library` to implement them sounds good.

> Also what about storybook? 

I'm not a big fan of `storybook` per-se but I think it might come in handy while developing the UI of the components.

> If you think this is a good idea @panz3r , I can create an issue for this, then we can start going through and refactoring bits of code like this

LGTM 👍

Reply:
@Mattcorner, I have opened #220 as a question to see if we want to look at the unused parameter in the filter function

Reply:
@mattcorner @panz3r  I have opened #221 Implement scaffolding for UI testing framework

Reply:
I have created Implement storybook #222